gammapy/estimators/flux.py

```diff
@@ -195,16 +160,20 @@ def run(self, datasets):
 
         model = self.get_scale_model(models)
 
+        energy_min, energy_max = datasets.energy_ranges
```

Comment:
This is unrelated to the PR directly but this method is returning something different from the `MapDataset.energy_range` which takes into account masks. This is confusing.

Reply:
Good catch, indeed we should harmonise and mirror the API from the `MapDataset` level on the `Datasets` level.